src/lib.rs

```diff
@@ -230,10 +222,7 @@ pub fn find_shebang(reader: impl io::Read) -> Option<String> {
     let mut buffered_reader = io::BufReader::new(reader);
 
     let mut line = String::new();
-    match buffered_reader.read_line(&mut line) {
-        Err(_) => return None,
-        _ => (),
-    };
+    buffered_reader.read_line(&mut line).ok()?;
```

Comment:
OK, I don't get how this works: [`BufRead.read_line()`](https://doc.rust-lang.org/std/io/trait.BufRead.html#method.read_line) -> [`Result`](https://doc.rust-lang.org/std/io/type.Result.html), [`Result.ok()`](https://doc.rust-lang.org/std/result/enum.Result.html#method.ok) -> `Option`. But [The Book](https://doc.rust-lang.org/book/second-edition/ch09-02-recoverable-errors-with-result.html#the--operator-can-only-be-used-in-functions-that-return-result) says that `?` can only be used in functions that return `Result`.

So what the heck is going on here? 😕